### PR TITLE
file literals

### PR DESCRIPTION
**Pull request by danbills**
From the Spec:

> If no location or path is specified, a file object must specify contents with the UTF-8 text content of the file. This is a "file literal". File literals do not correspond to external resources, but are created on disk with contents with when needed for a executing a tool. Where appropriate, expressions can return file literals to define new files on a runtime. The maximum size of contents is 64 kilobytes.

### CODE REVIEW

File: cwl/src/main/scala/cwl/CwlJsonToDelayedCoercionFunction.scala [hunk 1]
```diff
@@ -62,6 +62,20 @@ private [cwl] object CwlJsonToDelayedCoercionFunction extends Json.Folder[Delaye
...
+          This is a "file literal". File literals do not correspond to external resources, but are created on disk with
+          contents with when needed for a executing a tool. Where appropriate, expressions can return file literals to
+          define new files on a runtime. The maximum size of contents is 64 kilobytes.
+
+          As this is <= 64 KB of data and is being put in a temp dir, the author doesn't anticipate needing a formal cleanup.
+         */
+        case Right(file@File(_, None, None, _, _,_,_,_,Some(contents))) => {
+          val tempDir = better.files.File.newTemporaryDirectory()
+          val cwlFile: better.files.File = tempDir./(s"${contents.hashCode}").write(contents)
+          file.asWomValue.map(_.copy(valueOption = Some(cwlFile.path.toString)))
```
Comment (mcovarr) [inline cwl/src/main/scala/cwl/CwlJsonToDelayedCoercionFunction.scala]: `Option`

File: cwl/src/main/scala/cwl/CwlJsonToDelayedCoercionFunction.scala [hunk 2]
```diff
@@ -62,6 +62,20 @@ private [cwl] object CwlJsonToDelayedCoercionFunction extends Json.Folder[Delaye
...
       Json.fromJsonObject(value).as[File] match {
         case Left(errors) => errors.message.invalidNel
+        /*
+          From the CWL spec:
+          If no location or path is specified, a file object must specify contents with the UTF-8 text content of the file.
+          This is a "file literal". File literals do not correspond to external resources, but are created on disk with
+          contents with when needed for a executing a tool. Where appropriate, expressions can return file literals to
+          define new files on a runtime. The maximum size of contents is 64 kilobytes.
+
+          As this is <= 64 KB of data and is being put in a temp dir, the author doesn't anticipate needing a formal cleanup.
```
Comment (mcovarr) [inline cwl/src/main/scala/cwl/CwlJsonToDelayedCoercionFunction.scala]: the optimism is admirable

Comment (kshakir) [reply cwl/src/main/scala/cwl/CwlJsonToDelayedCoercionFunction.scala]: Heh. My 2¢: I'm ok with this for the moment as long as an issue is filed and referenced here in the comments, similar to [this one](https://github.com/broadinstitute/cromwell/issues/3186).

**Comment by danbills** [comment]
seems some additional file literal tests have appeared, going to make sure we pass those before merging

**Comment by danbills** [comment]
closing this temporarily as there's a significant amount of work to do w.r.t. cloud backends

**Pull request closed without merging**